File: services/user-service/src/main/java/com/visor/school/userservice/repository/ParentRepository.java
package com.visor.school.userservice.repository;

import java.util.List;
import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.visor.school.userservice.model.Parent;

@Repository
public interface ParentRepository extends JpaRepository<Parent, UUID> {
    List<Parent> findByUserId(UUID userId);
    List<Parent> findByStudentId(UUID studentId);
    boolean existsByUserIdAndStudentId(UUID userId, UUID studentId);
}

File: services/user-service/src/main/kotlin/com/visor/school/userservice/repository/ParentRepository.kt
package com.visor.school.userservice.repository

import com.visor.school.userservice.model.Parent
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface ParentRepository : JpaRepository<Parent, UUID> {
    fun findByUserId(userId: UUID): kotlin.collections.List<Parent>
    fun findByStudentId(studentId: UUID): kotlin.collections.List<Parent>
    fun existsByUserIdAndStudentId(userId: UUID, studentId: UUID): Boolean
}

